### ISSUE TITLE
/channels as a parameterized LIST command

### ISSUE THREAD

**Issue by eHammarstrom** [osa1/tiny#382]
To list channels I currently use LIST, but sometimes I want to change the ordering, or perhaps order by number of users instead.

```
/channels [ORDER_BY] [ORDERING]
  ORDER_BY: name (default), num_users
  ORDERING: ascending, descending
```

No args to the command would result in the default LIST behavior.

**Comment by sullivant** [comment]
I believe "LIST" is a server command - ie it will be sent to and processed by the server, not by tiny.

**Comment by sullivant** [comment]
So, I am still getting used to doing more "shared" rust - I have some personal projects and not much experience in the most idiomatic way of doing things (my day-to-day is java).  However I forked and created a rudimentary "channels" command in my commit here:

sullivant/tiny@cdac2f3

This does the basic, but it adds it as a TUI command as I was not really sure how to get a list of channel tabs (or any tabs really) while in the file `crates/tiny/src/cmd.rs`.  I still need to look into that if that's where it belongs, as well as, of course parameterizing the command and including nick counts within that chan. 

This is my first here - so feel free to toss this away if it's in a totally incorrect direction, but I'd like to learn more, that's for sure as I use tiny every day and love it.

**Comment by trevarj** [comment]
@sullivant I think you are on the right track. The implementation will need to be similar to how `names()` (`/names`) works in `cmd.rs`.

So from your new `channels()` function you will call `find_client()` and then within in the client you will send the `LIST` message and parse the result according to the inputs (order by, ordering).

Let me know if you need any help or find us in the #tiny channel on oftc.

**Comment by sullivant** [comment]
Gotcha - though that would possibly require a lot of returned data, no?  (There may be many many channels on a server.). The original post mentioned "that I use" (when listing channels), so I went that route.  

Going to take the advice re `/names` and run with it.  Thanks!

**Comment by trevarj** [comment]
> however there may be a lot of them

Yes, never run in a channel tab or you flood it!

**Comment by sullivant** [comment]
Certainly.  For instance if I'm connected to irc.libera.chat and run a `LIST` ... it takes quite a bit of time to get through it.  Content shows up looking pretty standard:

```
zirconium.libera.chat: flandish #zuul 11
zirconium.libera.chat: flandish #zynaddsubfx 10  The ZynAddSubFX open source musical synthesizer https://zynaddsubfx.sf.net | https://github.com/zynaddsubfx https://github.com/mruby-zest
zirconium.libera.chat: flandish #zypp 5
```

On this server, it's sorted by default alpha - which is fine, but there is very little ability to grep/grok any of it, as it all scrolls off.   I wonder if also a param for "filtering" would be nice to have - ie: `"min_users=10"` to only show the obvious, etc.  

Anyway, I've got a bit of time today, but have to get back to some for-pay work.  I hope to take a look at this more, and by all means, please understand that I'm enjoying this learning process, but don't wait for me if someone else already has it. :)

**Comment by trevarj** [comment]
> I wonder if also a param for "filtering" would be nice to have - ie: "min_users=10" to only show the obvious

Yeah, I think this is common among other clients, for example irssi offers some filtering by name and number of users

Ref: https://irssi.org/documentation/help/list/
